### ISSUE TITLE
Failing SF code with multiple vs statements

### ISSUE THREAD

**Issue by t-b** [AllenInstitute/MIES#1809]
```
1 vs 3

with

2 vs 4
```

creates the following plot which is wrong.

![image](https://github.com/AllenInstitute/MIES/assets/1185677/00b0d5c4-6195-416b-a851-1307326f827b)

```
[1] vs [a]
with
[2] vs [b]
```

fails in

```igorpro
		if(action != SF_ACTION_SKIP && lastAction == SF_ACTION_ARRAY)
			// If the last action was the handling of "]" from an array
			SFH_ASSERT(action == SF_ACTION_ARRAYELEMENT || action == SF_ACTION_HIGHERORDER, "Expected \",\" after \"]\"", jsonId=jsonId)
		endif
```

as action is `SF_ACTION_COLLECT`.

**Comment by t-b** [comment]
```
1 vs a

with

[2] vs b
```

fails as well.

**Comment by t-b** [comment]
```
range(1) vs wave(tt)

with

range(1) vs wave(tt)
```

gives `Only one operation allowed`.

**Comment by t-b** [comment]
The combination of `vs` and `with` is broken.

**Comment by t-b** [comment]
The bug here seems to be by-design.

In the documentation we have an example like (abridged):

```
   0...10
   with
   20...30
    vs range(10, 100, 10)
```

which produces

![grafik](https://github.com/AllenInstitute/MIES/assets/1185677/e0469af6-fbbb-48f3-84fa-7f662b04fae6)

So the current design defines `vs` to be for both formulas of `with`.
But I was here under the impression that `vs` only binds to a single formula i.e. (not valid SF)

```
   0...10
   with 
   (
   20...30
    vs range(10, 100, 10)
	)
```

@timjarsky Is the current design what you would expect?

**Comment by timjarsky** [comment]
@t-b I'm not sure I can say I expected that result, but it is convenient. I can also see why it would be problematic because being explicit does not work, e.g.,

```
0...10
vs range(1, 10, 1)
 
with 
   
20...30
vs range(10, 100, 10)
```

gives"One operation only is allowed" error

**Comment by timjarsky** [comment]
@t-b I think I'm leaning your way as I am realizing that any convenience gained comes with a cost for a use case that I think is more common.

**Comment by MichaelHuth** [comment]
I think the background here is, that the last example would require the formula plotter to support two different x-axes in the same plot.

**Comment by timjarsky** [comment]
> > I think the background here is, that the last example would require the formula plotter to support two different x-axes in the same plot.
> 
> Well the data does share an x-axis, just the ranges are different.

Right, we don't want to support two different x-axes, but instead, recognize when they can share an axis.

**Comment by t-b** [comment]
So I've spoke with @MichaelHuth today about this and we think we need a new keyword to distinguish an x-axis for all formula and an x-axis per formula.

Idea:
- Introduce `allvs` as keyword for setting the x-axis for all y-formulas for a graph.
- Redefine `vs` to be only allowed for per-formula usage
- Allow `vs []` to override a global x-axis defined via `allvs`

New working examples:

```
0...10
vs range(1, 10, 1)
 
with 
   
20...30
vs range(10, 100, 10)
```

```
[1]

with

[2] vs [] # override global x-axis and prefer data's SetScale

with

[3] vs [4] # override global x-axis

allvs [1]
```

We very likely need this bug resolved somehow for the SF plotting of #1532.
